rosa_sts_managed_oidc/main.tf

```diff
@@ -0,0 +1,97 @@
+module "vpc" {
+  create_vpc           = var.create_vpc
+  source               = "../modules/network"
+  aws_region           = var.aws_region
+  vpc_name             = var.vpc_name
+  additional_tags      = var.additional_tags
+  vpc_cidr_block       = var.vpc_cidr_block
+  availability_zones   = var.availability_zones
+  private_subnet_cidrs = var.private_subnet_cidrs
+  public_subnet_cidrs  = var.public_subnet_cidrs
+  single_nat_gateway   = var.single_nat_gateway
+}
+
+# Create the main account roles for ROSA
+module "account_role" {
+  create_account_roles   = var.create_account_roles
+  source                 = "../modules/account_roles"
+  account_role_prefix    = var.account_role_prefix
+  path                   = var.path
+  ocm_environment        = var.ocm_environment
+  rosa_openshift_version = var.rosa_openshift_version
+  account_role_policies  = var.account_role_policies
+  all_versions           = var.all_versions
+  operator_role_policies = var.operator_role_policies
+  additional_tags        = var.additional_tags
+}
+
+# This is unfortunately needed as the cluster will try to build before the account role
+# is created. Hopefully this is fixed in the future
+resource "time_sleep" "wait_10_seconds" {
+  depends_on = [module.account_role]
+
+  create_duration = "10s"
+}
+
+# Create managed OIDC config
+module "oidc_provider" {
+  source               = "../modules/managed_oidc_provider"
+  operator_role_prefix = var.operator_role_prefix
+  account_role_prefix  = var.account_role_prefix
+  additional_tags      = var.additional_tags
+  path                 = var.path
+}
+
+# Create the operator roles for ROSA
+module "operator_roles" {
+  create_operator_roles = var.create_operator_roles
+  source               = "../modules/operator_roles"
+  oidc_thumbprint       = module.oidc_provider.thumbprint
+  oidc_endpoint_url     = module.oidc_provider.oidc_endpoint_url
+  operator_role_prefix = var.operator_role_prefix
+  additional_tags      = var.additional_tags
+}
+
+# Make the ROSA Cluster
+module "rosa_cluster" {
+  source                 = "../modules/rosa_cluster"
+  cluster_name           = var.cluster_name
+  rosa_openshift_version = var.rosa_openshift_version
+  aws_region             = var.aws_region
+  multi_az               = var.multi_az
+  availability_zones     = var.availability_zones
+  account_role_prefix    = var.account_role_prefix
+  operator_role_prefix   = var.operator_role_prefix
+  machine_type           = var.machine_type
+  proxy                  = var.proxy
+  autoscaling_enabled    = var.autoscaling_enabled
+  min_replicas           = var.min_replicas
+  max_replicas           = var.max_replicas
+  oidc_config_id         = module.oidc_provider.id
+  additional_tags        = var.additional_tags
+  vpc_cidr_block         = var.vpc_cidr_block
+
+  #private link cluster values
+  enable_private_link = var.enable_private_link
+  aws_subnet_ids   = var.create_vpc ? module.vpc.private_subnets : var.aws_subnet_ids
```

Comment:
you should set also public subnets.... not only private...

Reply:
Currently this isn't working with public clusters. It's something that needs to be edited. I've been focusing on Black Belt customer use cases which are almost 100% of the time private clusters. Once I have some free time I'll come back to this and add tests for public, or someone else is more than welcome to add this in with the appropriate check if the cluster is not private :-)